openhcl/host_fdt_parser/src/lib.rs

```diff
@@ -514,6 +517,16 @@ impl<
 
                                 storage.entropy = Some(entropy);
                             }
+                            // These parameters may not be present so it is not an error if they are missing.
+                            "servicing" => {
+                                storage.preserve_dma_4k_pages =
+                                    match openhcl_child.find_property("dma-preserve-pages") {
```

Comment:
can be simplified to

```rust
openhcl_child.find_property("dma-preserve-pages").ok().map(|pages| ...)
```

Reply:
Do you agree with the idea to have both keys optional? If yes then I probably should leave the current as is, if no - then I'll change. Please let me know.

Reply:
this suggestion (and the use of `and_then` from the other comment) preserve the double-optional semantics. they are purely syntactic simplifications